File: test/yosupo-enumerate-quotients.test.cpp
#define PROBLEM "https://judge.yosupo.jp/problem/enumerate_quotients"

#include "../cpp/sqrt.hpp"
#include <vector>
#include <algorithm>
#include <iostream>

int main() {
    long long n;
    std::cin >> n;
    std::vector<long long> v1, v2;
    long long m = sqrtll(n);
    for(long long i = 1; i <= m; i ++) {
        v1.push_back(n / i);
        v2.push_back(i);
    }
    if(v1.back() == v2.back()) v1.pop_back();

    std::cout << v1.size() + v2.size() << std::endl;
    reverse(v1.begin(), v1.end());
    for(long long e : v2) std::cout << e << " ";
    for(long long e : v1) std::cout << e << " ";
    std::cout << std::endl;

    return 0;
}

File: test/yosupo-enumerate-quotients.test.py
# verification-helper: PROBLEM https://judge.yosupo.jp/problem/enumerate_quotients
import sys
input = sys.stdin.buffer.readline

from py.sqrt import isqrt


def main() -> None:
    n = int(input())
    v1 = []
    v2 = []
    m = isqrt(n)
    for i in range(1, m+1):
        v1.append(n//i)
        v2.append(i)
    if(v1[-1] == v2[-1]):
        v1.pop()
    v1.reverse()
    print(len(v1)+len(v2))
    print(*(v2+v1))


if __name__ == "__main__":
    main()
